src/metabase/public_settings.clj

```diff
@@ -943,8 +895,64 @@ See [fonts](../configuring-metabase/fonts.md).")
   :default    true
   :type       :boolean)
 
+(defsetting query-analysis-force-synchronicity
+  (deferred-tru "Whether query analysis should happen synchronously or not. Enabling this is almost certainly a bad
+  idea with respect to performance; the setting exists to avoid race conditions when testing.")
+  :visibility :admin ; not :internal so that it can be toggled in E2E tests
```

Comment:
I like the idea of the tests avoiding any "funny business", but there are two downsides to this visibility.

1. Customer visible, means we need to be more careful.
2. By publicly exposing the setting we potentially make it harder to iterate on the scheduling mechanism.
3. The tests need to do a bunch of navigation in each test before starting.

Maybe this is still the best way, but as a devil's advocate I wonder whether we couldn't just have the E2E tests use a command line argument or update the setting directly in the db. The latter might not work if the app is caching the setting in memory though.